src/MagnumPlugins/UfbxImporter/UfbxImporter.cpp

```diff
@@ -0,0 +1,1287 @@
+}
+
+Int UfbxImporter::doTextureForName(Containers::StringView name) {
+    const ufbx_element* element = ufbx_find_element_len(_state->scene.get(), UFBX_ELEMENT_TEXTURE, name.data(), name.size());
+    return element ? _state->textureRemap[element->typed_id] : -1;
+}
+
+Containers::String UfbxImporter::doTextureName(UnsignedInt id) {
+    const FileTexture& fileTexture = _state->textures[id];
+    return _state->scene->textures[fileTexture.textureIndex]->name;
+}
+
+Containers::Optional<TextureData> UfbxImporter::doTexture(UnsignedInt id) {
+    const FileTexture& fileTexture = _state->textures[id];
+    const ufbx_texture* texture = _state->scene->textures[fileTexture.textureIndex];
+
+    const SamplerWrapping wrappingU = toSamplerWrapping(texture->wrap_u);
+    const SamplerWrapping wrappingV = toSamplerWrapping(texture->wrap_v);
+    return TextureData{TextureType::Texture2D,
+        SamplerFilter::Linear, SamplerFilter::Linear, SamplerMipmap::Linear,
+        {wrappingU, wrappingV, SamplerWrapping::ClampToEdge},
+        fileTexture.fileTextureIndex};
+}
+
+AbstractImporter* UfbxImporter::setupOrReuseImporterForImage(UnsignedInt id, const char* errorPrefix) {
+    const ufbx_texture_file& file = _state->scene->texture_files[id];
+
+    /* Looking for the same ID, so reuse an importer populated before. If the
+       previous attempt failed, the importer is not set, so return nullptr in
+       that case. Going through everything below again would not change the
+       outcome anyway, only spam the output with redundant messages. */
+    if(_state->imageImporterId == id)
+        return _state->imageImporter ? &*_state->imageImporter : nullptr;
+
+    /* Otherwise reset the importer and remember the new ID. If the import
+       fails, the importer will stay unset, but the ID will be updated so the
+       next round can again just return nullptr above instead of going through
+       the doomed-to-fail process again. */
+    _state->imageImporter = Containers::NullOpt;
+    _state->imageImporterId = id;
+
+    AnyImageImporter importer{*manager()};
+    importer.setFlags(flags());
+    if(fileCallback()) importer.setFileCallback(fileCallback(), fileCallbackUserData());
+
+    if(file.content.size > 0) {
+        auto textureData = Containers::ArrayView<const char>(reinterpret_cast<const char*>(file.content.data), file.content.size);
+        if(!importer.openData(textureData))
+            return nullptr;
+    } else if(file.filename.length > 0) {
+        if(!_state->fromFile && !fileCallback()) {
+            Error{} << errorPrefix << "external images can be imported only when opening files from the filesystem or if a file callback is present";
+            return nullptr;
+        }
+
+        if(!importer.openFile(file.filename))
+            return nullptr;
+    } else {
+        Error{} << errorPrefix << "skipping external image defined by absolute path:" << Containers::StringView(file.absolute_filename);
+        return nullptr;
```

Comment:
I thought it would fail fast as a warning that you're making fragile files that work only on your filesystem, though now thinking about it maybe it's not the responsibility of the plugin :D (maybe a warning at worst)

Reply:
I'm not really dealing with this case anywhere else, and doing it for just one plugin out of all seems inconsistent for me.

In my opinion the importer should just accept that as a sad truth, same as with material/image files referenced from OBJs sometimes mixing up forward and backward slashes or [having trailing spaces](https://github.com/mosra/magnum-plugins/pull/58).

Reply:
Fixed it to just load from the absolute paths if a relative one is not provided. ufbx tries it's best to normalize the file paths already, except absolute paths are kept as is as getting something like `\home\user\file.png` or `C:/Users/user/file.png` aren't really great.